modules/wpml/wpml-config.php

```diff
@@ -211,23 +224,143 @@ public function copy_post_metas( $metas, $sync ) {
 	public function copy_term_metas( $metas, $sync ) {
 		foreach ( $this->xmls as $xml ) {
 			$cfs = $xml->xpath( 'custom-term-fields/custom-term-field' );
-			if ( is_array( $cfs ) ) {
-				foreach ( $cfs as $cf ) {
-					$attributes = $cf->attributes();
-					if ( empty( $attributes ) ) {
-						continue;
-					}
-					if ( 'copy' == $attributes['action'] || ( ! $sync && in_array( $attributes['action'], array( 'translate', 'copy-once' ) ) ) ) {
-						$metas[] = (string) $cf;
-					} else {
-						$metas = array_diff( $metas, array( (string) $cf ) );
-					}
+
+			if ( ! is_array( $cfs ) ) {
+				continue;
+			}
+
+			foreach ( $cfs as $cf ) {
+				$action = $this->get_field_attribute( $cf, 'action' );
+
+				if ( 'copy' === $action || ( ! $sync && in_array( $action, array( 'translate', 'copy-once' ), true ) ) ) {
+					$metas[] = (string) $cf;
+				} else {
+					$metas = array_diff( $metas, array( (string) $cf ) );
 				}
 			}
 		}
+
 		return $metas;
 	}
 
+	/**
+	 * Adds post meta keys to export.
+	 *
+	 * @since 3.3
+	 * @see   PLL_Export_Metas
+	 *
+	 * @param  array $keys {
+	 *     A recursive array containing nested meta sub-keys to translate.
+	 *     Ex: array(
+	 *      'meta_to_translate_1' => 1,
+	 *      'meta_to_translate_2' => 1,
+	 *      'meta_to_translate_3' => array(
+	 *        'sub_key_to_translate_1' => 1,
+	 *        'sub_key_to_translate_2' => array(
+	 *             'sub_sub_key_to_translate_1' => 1,
+	 *         ),
+	 *      ),
+	 *    )
+	 * }
+	 * @return array
+	 *
+	 * @phpstan-param array<string, mixed> $keys
+	 * @phpstan-return array<string, mixed>
+	 */
+	public function post_metas_to_export( $keys ) {
+		// Add keys that have the `action` attribute set to `translate`.
+		foreach ( $this->xmls as $xml ) {
+			$fields = $xml->xpath( 'custom-fields/custom-field' );
+
+			if ( ! is_array( $fields ) ) {
+				// No custom fields.
+				continue;
+			}
+
+			foreach ( $fields as $field ) {
+				$action = $this->get_field_attribute( $field, 'action' );
+
+				if ( 'translate' !== $action ) {
+					continue;
+				}
+
+				$keys[ (string) $field ] = 1;
+			}
+		}
+
+		// Deal with sub-field translations.
+		foreach ( $this->xmls as $xml ) {
+			$fields = $xml->xpath( 'custom-fields-texts/key' );
+
+			if ( ! is_array( $fields ) ) {
+				// Wrong configuration: no `<key>` nodes.
```

Comment:
For me, there's nothing wrong, as we don't always need 'custom-fields-texts' (when meta are single strings). 
```suggestion
				// No 'custom-fields-texts` nodes.
```